tests/modules/programs/emacs/extra-config.nix

```diff
@@ -0,0 +1,25 @@
+{ config, lib, pkgs, ... }:
+
+let
+  testScript = pkgs.writeText "test.el" ''
+    ;; Emacs won't automatically load default.el when --script is specified
+    (load "default")
+    (kill-emacs (if (eq hm 'home-manager) 0 1))
+  '';
+
+  emacsBin = "${config.programs.emacs.finalPackage}/bin/emacs";
+in {
+  config = {
+    programs.emacs = {
+      enable = true;
+      extraConfig = "(setq hm 'home-manager)";
+    };
+
+    # running emacs with --script would enable headless mode
+    nmt.script = ''
+      if ! ${emacsBin} --script ${testScript}; then
+        fail "Failed to load default.el."
+      fi
+    '';
+  };
+}
```

Comment:
How big of a dependency is Emacs? (considering the CI and users will pull Emacs when running tests)

Reply:
It was 1.3 GB on NixOS, which was quite bigger than I expected. Using `emacs-nox` would reduce it to 581.3 MB but still moderately large.

I've put the test behind an `mkIf false`, because it's not trivial to test `emacsWithPackages` without using emacs.

Reply:
This is fairly big but the CI took 4 minutes overall so I think we could preserve these tests on the CI.

@rycee I think we could just add `ci ? false` to https://github.com/nix-community/home-manager/blob/029767ca63fe33d9dd002093fc065ea8fde565ab/tests/default.nix#L1
and override the CI to run tests with `nix-shell --arg ci true --pure tests -A run.all`

Reply:
I've now added `skipBig ? false`.